services/common/src/components/forms/RenderTreeSelect.tsx

```diff
@@ -0,0 +1,147 @@
+import { BaseInputProps, getFormItemLabel } from "./BaseInput";
+import { FormConsumer, IFormContext } from "./FormWrapper";
+import { EMPTY_FIELD } from "@mds/common/constants/strings";
+
+/**
+ * @constant RenderTreeSelect - Ant Design `TreeSelect` component for redux-form - used for hierarchical data sets;
+ * https://4x.ant.design/components/tree-select/
+ */
+
+interface RenderTreeSelectProps extends BaseInputProps {
+    loading?: boolean;
+    viewDisplay?: (opts: any) => ReactNode;
+}
+
+const defaultViewDisplay = (opts: any[]): ReactNode => {
+
+    if (!opts.length) {
+        return <div className="tree-select-view">
+            <Typography.Paragraph className="view-item-value">
+                {EMPTY_FIELD}
+            </Typography.Paragraph>s
+        </div>
+    }
+    return <ul className="tree-select-view margin-large--left">
+        {opts.map((opt) =>
+            <li key={opt.value ?? opt.id}>
+                <Typography.Paragraph className="view-item-value">{opt.title}</Typography.Paragraph>
+            </li>
+        )}
+    </ul>
+};
+
+const getLabelsRecursive = (value: any[], opts: any[]) => {
+    if (!opts?.length) { return []; }
+
+    const childLabels = opts.reduce((acc, o) => {
+        const oLabels = getLabelsRecursive(value, o.children);
+        return [...acc, ...oLabels]
+    }, []);
+
+    const includedNodes = opts.filter((o) => value.includes(o.value)).map((o) => ({ title: o.title, value: o.value }));
+    return [...includedNodes, ...childLabels]
+}
+
+export const parseTreeValues = (values: any[] | { value: any }[] | { id: any }[]) => {
+    if (values.length > 0) {
+        if (typeof values[0] === "object" && "value" in values[0]) {
+            return values.map((v: any) => v.value);
+        }
+        if (typeof values[0] === "object" && "id" in values[0]) {
+            return values.map((v: any) => v.id);
+        }
+    }
+    return values;
+};
+
+export const RenderTreeSelect: FC<RenderTreeSelectProps & TreeSelectProps> = ({
+    placeholder = "",
+    // treeData & treeDataSimpleMode- SEE antd documentation, but essentially 
+    // treeDataSimpleMode can be used to pass in an array of options with id references to parent nodes,
+    // or treeData can be used to pass in nested data. node data has more available properties such as checkable
+    treeData = [],
+    treeDataSimpleMode = false,
+    treeCheckStrictly = true, // don't associate parent/child
+    treeCheckable = true,
+    treeLine = true,
+    disabled = false,
+    label = "",
+    multiple = false,
+    treeDefaultExpandAll = true,
+    viewDisplay = defaultViewDisplay,
+    showNA,
+    meta,
+    input,
+    ...props
+}) => {
+    return (
+        <FormConsumer>
+            {(value: IFormContext) => {
+                const { isEditMode, isModal } = value;
+                if (!isEditMode) {
+                    let selectedOptions = [];
+                    if (treeDataSimpleMode) {
```

Comment:
I feel like I should drop a warning here that I have not tested treeDataSimpleMode at all.